pip_audit/cli.py

```diff
@@ -155,7 +178,9 @@ def audit():
     output_desc = args.desc.to_bool(args.format)
     formatter = args.format.to_format(output_desc)
 
-    with AuditSpinner() as state:
+    with ExitStack() as stack:
+        state = stack.enter_context(AuditSpinner()) if args.progress_spinner.to_bool() else None
```

Comment:
This is necessary in order to use the scoped enter/exit approach conditionally. I think it's still useful to make sure we `finalize`, but I could be convinced otherwise.

Reply:
TIL about `ExitStack`! Perhaps we could push this one level further -- instead of having `AuditState` be an ABC, it could manage an `ExitStack` whose members conform to an ABC/Subtype protocol. I'll fiddle around with that, but this LGTM for now.